### ISSUE TITLE
App Crashing 

### ISSUE THREAD

**Issue by seniorm0ment** [UnicornsOnLSD/finamp#199]
I notice fairly often I can throw something on then after a song or two the app just closes. It disappears from notifications menu and stops playing music.

Android 10, F-Droid Latest Version

**Comment by UnicornsOnLSD** [comment]
Is music playing when this happens? I've never seen anything like this before.

**Comment by Maxplosion** [comment]
sounds like very aggressive battery optimization that closes the app when it runs in the background maybe?
many phones have/had that and on some even adding apps to the whitelist does not work all the time. (looking at you Honor)

**Comment by seniorm0ment** [comment]
Yes music is always playing when it happens, and I have disabled any battery optimization for the app already.

**Comment by UnicornsOnLSD** [comment]
Which ROM do you use?

**Comment by seniorm0ment** [comment]
LG's stock (custom) Android 10 rom

**Comment by UnicornsOnLSD** [comment]
I can't find anything online saying that LG is particularly bad at killing apps. The app uses about 200-300MB of RAM when playing music, and Dart/Kotlin don't have memory leaks (unless there's a bug in the language itself or something). If you can get access to a logcat, try searching for finamp or flutter, there may be a crash report there if it really is hard crashing.

**Comment by seniorm0ment** [comment]
Yeah I'll check the logcat when I get some time

**Comment by seniorm0ment** [comment]
I haven't had time to check, but I'm wondering if I'm actually getting thrown for a loop, and it's just due to the song ending and nothing else being in the queue, so the notification disappears. I did notice this happen yesterday.
The queue does seem to display wonky behavior, and the shuffle button for albums does not work. I actually don't seem to be able to add stuff to queue then view the list of it either idk.

I'll keep a lookout and see if I notice whether it does it during a song playback, or at the end.

**Comment by UnicornsOnLSD** [comment]
Yeah, the notification will disappear when the queue ends, although the new release will be more like how it "should" be with the queue returning to the start once its done (personally I prefer the notification disappearing but Android 11 likes players to persist). Album shuffling should work, but it always picks the first song. This is a known issue and is tracked at #61, I'm pretty sure it's an issue with the player package I use.

**Comment by Chaphasilor** [comment]
@seniorm0ment is this still an issue for you?

**Comment by seniorm0ment** [comment]
I have not noticed any crashing personally
